File: js/1011.js
const fs = require('fs');

const inputs = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const t = +inputs[0];

for (let tc = 1; tc <= t; tc++) {
  const [x, y] = inputs[tc].split(' ').map(Number);
  let distance = y - x;
  let answer;

  const sqrtValue = Math.sqrt(distance);
  if (sqrtValue % 1 === 0) {
    answer = sqrtValue * 2 - 1;
  } else {
    const a = Math.ceil(sqrtValue);
    const top = a ** 2;
    const bottom = (a - 1) ** 2;
    const middle = bottom + (top - bottom - 1) / 2;

    if (distance <= middle) answer = a * 2 - 2;
    else answer = a * 2 - 1;
  }

  console.log(answer);
}

File: c++/1011.cpp
#include <cmath>
#include <iostream>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  int t;
  cin >> t;
  for (int tc = 0; tc < t; tc++) {
    long long x, y;
    cin >> x >> y;
    long long distance = y - x;
    long long answer;

    long long i = 1;
    while (i * i <= distance) {
      i++;
    }
    i -= 1;

    if (i * i == distance) {
      answer = i * 2 - 1;
    } else {
      long long bottom = i * i;
      long long top = (i + 1) * (i + 1);
      long long middle = bottom + (top - bottom - 1) / 2;

      if (distance <= middle)
        answer = 2 * i;
      else
        answer = 2 * i + 1;
    }
    cout << answer << '\n';
  }

  return 0;
}